src/hooks/useForceSync.js

```diff
@@ -0,0 +1,35 @@
+import { useCallback } from 'react'
+
+function useForceSync({ doc, serviceClient, collabProvider }) {
+  const noteId = doc && doc.file.id
+
+  // Be sure to save everything before leaving
+  // and force sync with the io.cozy.file
+  const forceSync = useCallback(
+    async function forceSync() {
+      if (doc && collabProvider && serviceClient) {
+        // wait for every event to finish
+        const provider = await collabProvider.provider
+        await provider.channel.ensureEmptyQueue()
+        // then force a server sync
+        await serviceClient.sync(noteId)
+      }
+    },
+    [doc, collabProvider]
+  )
+
+  // Sync on unload will probably be stopped by the browser,
+  // as most async code on unload, but let's try anyway
+  const emergencySync = useCallback(
+    function() {
+      if (doc && serviceClient) {
+        serviceClient.sync(noteId) // force a server sync
+      }
+    },
+    [doc, serviceClient]
+  )
+
+  return { forceSync, emergencySync }
+}
```

Comment:
Should forceSync and emergencySync be methods on the service ?

I think that there should be the least amount of code inside hooks, so that they can live outside React.

Reply:
En fait le fonctionnel est déjà dans `serviceClient.sync` et `channel.ensureEmptyQueue`. Ces deux méthodes gèrent l'interaction avec les états du composant (est ce que le doc est chargé, est-ce que les services sont prêts)